packages/kotti-ui/source/kotti-field-file-upload/validators.ts

```diff
@@ -0,0 +1,102 @@
+import {
+	KottiFieldFileUpload,
+	KottiFieldFileUploadRemote,
+	Shared,
+} from './types'
+
+export const isSelectingMultipleFilesWhenNotAllowed = (
+	allowMultiple: boolean,
+	selectionSize: number,
+): boolean => !allowMultiple && selectionSize > 1
+
+const isValidFileSize = (fileSize: number, maxFileSize: number): boolean =>
+	fileSize <= maxFileSize
+
+const isValidFileExtension = (
+	fileName: string,
+	validFileExtensions: Array<string>,
+): boolean => {
+	if (validFileExtensions.length === 0) return true
+
+	const fileExtension = fileName.split('.').pop() ?? null
+
+	if (fileExtension === null) return false
+
+	return validFileExtensions.some(
+		(extension) =>
+			extension.localeCompare(fileExtension.trim(), undefined, {
+				sensitivity: 'accent',
+			}) === 0,
+	)
+}
+
+/**
+ * Validates a File Item
+ * @param param0.extensions Allowed file extensions
+ * @param param0.fileName The name of the file (muss include file extension)
+ * @param param0.fileSize The size of the file in bytes
+ * @param param0.maxFileSize The maximum file size allowed in bytes
+ * @returns Validation.SUCCESS if the file is valid, else either
+ * Validation.INVALID_EXTENSION or Validation.MAX_SIZE_EXCEEDED.
+ * Validation.INVALID_EXTENSION has higher priority.
+ */
+export const validateFile = ({
+	extensions,
+	fileName,
+	fileSize,
+	maxFileSize,
+}: {
+	extensions: KottiFieldFileUpload.PropsInternal['extensions']
+	fileName: string
+	fileSize: number
+	maxFileSize: KottiFieldFileUpload.PropsInternal['maxFileSize']
+}): Shared.Validation => {
+	if (!isValidFileExtension(fileName, extensions))
+		return Shared.Validation.INVALID_EXTENSION
+
+	if (!isValidFileSize(fileSize, maxFileSize))
+		return Shared.Validation.MAX_SIZE_EXCEEDED
+
+	return Shared.Validation.SUCCESS
+}
+
+/**
+ * Checks whether Value is valid
+ * @param value The value to evaluate
+ * @returns true if at least one selected file is valid, else false
+ */
+export const isValidValue = <
+	T extends
+		| KottiFieldFileUpload.ValueInternal
+		| KottiFieldFileUploadRemote.ValueInternal,
+>(
+	value: T,
+	type: T extends KottiFieldFileUpload.ValueInternal ? 'NonRemote' : 'Remote',
```

Comment:
did you end up using the generics for common functions only here

I'd keep it here if that's the only case but I'd otherwise move the 'NonRemote' and 'Remote' as a type in the types file

I recall we tried with an enum - if it doesn't work with enum, just make them two standalone types for similar usages

if it's ony used here - keep it as is.

Reply:
yes, it's used only here for `isValidValue` bc in both cases (remote, non-remote) the arguments it takes are the same 

for other cases like `buildFileInfo`, which takes different arguments, are separated